### PR TITLE
error handling for invalid family id update

### PR DESCRIPTION
**Pull request by hanars**

### CODE REVIEW

File: seqr/views/apis/family_api.py [hunk 1]
```diff
@@ -41,12 +41,44 @@ def edit_families_handler(request, project_guid):
...
+    if family_guids:
+        family_models.update({f.guid: f for f in Family.objects.filter(project=project, guid__in=family_guids)})
+        if len(family_models) != len(family_guids):
+            missing_guids = set(family_guids) - set(family_models.keys())
+            return create_json_response({'error': 'Invalid family guids: {}'.format(', '.join(missing_guids))}, status=400)
+
+        updated_family_ids = {
+            fields[FAMILY_ID_FIELD]: family_models[fields['familyGuid']].family_id for fields in modified_families
+            if fields.get('familyGuid') and fields.get(FAMILY_ID_FIELD) and \
+                fields[FAMILY_ID_FIELD] != family_models[fields['familyGuid']].family_id}
```
Comment (ShifaSZ) [inline seqr/views/apis/family_api.py:55]: The `family_models[fields['familyGuid']]` could be undefined when the family with a GUID of `fields['familyGuid']` couldn't be found. Can this case happen?

File: seqr/views/apis/family_api.py [hunk 2]
```diff
@@ -41,12 +41,44 @@ def edit_families_handler(request, project_guid):
         return create_json_response(
             {}, status=400, reason="'families' not specified")
 
+    family_guids = [f['familyGuid'] for f in modified_families if f.get('familyGuid')]
```
Comment (ShifaSZ) [inline seqr/views/apis/family_api.py:44]: Would it be cleaner by using the code below?
```
family_guids = [x for x in [f.get('familyGuid') for f in modified_families] if x is not None]
```

**Review by ShifaSZ — CHANGES_REQUESTED** [review]

File: seqr/views/apis/family_api.py (hunk 2 above)
Comment (hanars) [reply seqr/views/apis/family_api.py:44]: no, a single list comprehension is much faster and easier to read the a nested double list comprehension

File: seqr/views/apis/family_api.py (hunk 1 above)
Comment (ShifaSZ) [reply seqr/views/apis/family_api.py:55]: I mean if `family_models[fields['familyGuid']]` is undefined or None, then `family_models[fields['familyGuid']].family_id` will be an error.

Comment (hanars) [reply seqr/views/apis/family_api.py:55]: there is error handling 5 lines above this that returns an error to the user if there isn't a model for any of the guids, so thats completely impossible

Comment (ShifaSZ) [reply seqr/views/apis/family_api.py:55]: Okay, I see what you mean.

**Review by ShifaSZ — APPROVED** [review]

**Pull request merged**